src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/CallGCSuppressFinalizeCorrectlyTests.cs

```diff
@@ -135,6 +135,47 @@ End Sub
             await VerifyVB.VerifyAnalyzerAsync(code);
         }
 
+        [Fact]
+        public async Task AsyncDisposableWithFinalizer_CSharp_NoDiagnostic()
+        {
+            var code = @"
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+class MyAsyncDisposable : IAsyncDisposable
+{
+    [DllImport(""example.dll"")]
+    private static extern int GetHandle();
+
+    [DllImport(""example.dll"")]
+    private static extern void FreeHandle(int handle);
+
+    private readonly int handle;
+
+    public MyAsyncDisposable()
+    {
+        this.handle = GetHandle();
+    }
+
+    ~MyAsyncDisposable()
+    {
+        FreeHandle(this.handle);
+    }
+
+    public async ValueTask DisposeAsync()
+    {
+        await Task.Run(() => FreeHandle(this.handle)).ConfigureAwait(false);
+        GC.SuppressFinalize(this);
+    }
+}";
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
```

Comment:
Build fails on this line. @mavasani Can you help clarifying the right way to write this test?

I got this line from another test, not sure why it works there but not here.

Reply:
You are missing the `using Test.Utilities;` and that should be ok.

Reply:
@Evangelink Thanks!
I've added it and waiting for the build to finish.

Reply:
@Evangelink, That `using` directive fixed the compilation error. But I can't get the test to pass.